### DISCUSSION TITLE
Handshake domain support

### DISCUSSION THREAD

**Discussion by hhkelder** [vercel/community#2426] [Idea]
The Handshake blockchain is a decentralized, permissionless blockchain protocol designed to enable the creation, registration, and transfer of internet domain names. It aims to provide a more secure, censorship-resistant alternative to traditional domain name systems (DNS).

Handshake uses a Proof-of-Work (PoW) consensus mechanism, similar to Bitcoin, to secure its blockchain network. However, instead of relying on a central authority, such as a domain registrar, Handshake allows users to register and manage domain names directly on the blockchain.

One of the key features of Handshake is its decentralized naming system, which replaces the hierarchical structure of traditional DNS. With Handshake, domain names are stored and resolved on the blockchain, making it more resistant to censorship and control by centralized authorities.

To register a domain name on the Handshake blockchain, users participate in a unique auction process. The bidding involves revealing cryptographic proofs and competing for ownership of a domain name. The winning bidder obtains control of the domain for a specified period.

Handshake aims to create a more open and equitable internet by decentralizing the domain name system and giving ownership and control back to individual users. It offers an alternative approach to domain registration and management that challenges the existing centralized DNS infrastructure.

Vercel is one of the few hosting providers that allow users to assign Handshake domains to their projects. The only issue is that there is a red notice saying "incorrect configuration", presumably because the domains are not using ICANN top level domains. 

It would be absolutely amazing if Vercel could remove that red notice, or better yet, officially support Handshake.

**Comment by smaeda-ks** [comment]
@hhkelder Are you talking about `hns.to` eTLD?

**Comment by hhkelder** [comment]
`hns.to` is related. It is a gateway for handshake domains created by a community member afaik. 

ie.  a handshake domain could be `henry.kelder`.   But since most browsers don't support handshake domains, you can access the domain on any browser by using `henry.kelder.hns.to`. 

My original message was regarding support for handshake domains directly. 

The thing is that they _already_ work in Vercel, the configuration is just not considered valid. 

I've attached a screenshot of what I mean.

`h.kelder` is a handshake domain. I've attached it to my project, and have been provided a working IP address. The website can be resolved in a handshake supported browser, but there is a red notice telling me my configuration is incorrect. 

![Screenshot 2023-05-12 at 10 30 52](https://github.com/vercel/community/assets/10614464/5c3d8f6c-8c99-4f0f-8a51-18572a6df7d1)

**Reply by smaeda-ks** [reply]
@hhkelder That message can be ignored. It is mainly there because we automatically issue Let's Encrypt certificates by default.

So if `h.kelder` is resolving to Vercel IPs from supported environments, it should reach Vercel deployments. But the problem is we always enforce HTTPS, and we only allow uploading customer certs for Enterprise customers, so I assume you always end up with a TLS error anyway.
